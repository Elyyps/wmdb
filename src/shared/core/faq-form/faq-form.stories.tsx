import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { FAQFormComponent } from "./faq-form.component";

storiesOf("FAQForm", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <FAQFormComponent onSubmit={() => ""} />);
