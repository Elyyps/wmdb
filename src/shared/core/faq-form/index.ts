export * from "./faq-form.component";
