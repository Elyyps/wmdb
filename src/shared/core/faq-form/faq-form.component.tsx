import * as React from "react";
import styles from "./faq-form-component.module.scss";
import { TextAreaComponent } from "@app/core/text-area";
import { withFormik, FormikProps, FormikErrors, Form } from "formik";
import { validateEmail } from "@app/util/validate-email";
import { Button } from "../button";
import { Input } from "../input";

interface IFAQFormErrorMessages {
  emailAddress: string;
  name: string;
}
export interface IFAQFormValues {
  emailAddress: string;
  name: string;
}
const InnerForm = (props: FormikProps<IFAQFormValues>) => {
  const { touched, errors } = props;

  return (
    <div className={styles["faq-form"]}>
      <Form action="#">
        <Input
          label={"Naam"}
          name={"name"}
          errorMessage={touched.name ? errors.name : ""}
          onChange={props.handleChange}
          onBlur={props.handleBlur}
          value={props.values.name}
        />
        <br />
        <Input
          label={"E-mailadres"}
          name={"emailAddress"}
          errorMessage={touched.emailAddress ? errors.emailAddress : ""}
          onChange={props.handleChange}
          onBlur={props.handleBlur}
          value={props.values.emailAddress}
        />
        <TextAreaComponent label={"Vraag of opmerking (optioneel)"} name={"textarea"} />
        <div className="form__item">
          <Button title={"Verstuur mijn vraag"} variant={"secondary"} fullWidth />
        </div>
      </Form>
    </div>
  );
};
interface IFormProps {
  onSubmit: (values: IFAQFormValues) => void;
}
export const FAQFormComponent = withFormik<IFormProps, IFAQFormValues>({
  mapPropsToValues: () => ({
    emailAddress: "",
    name: ""
  }),

  validate: (values: IFAQFormValues) => {
    const errors: FormikErrors<IFAQFormErrorMessages> = {};

    if (!values.name) {
      errors.name = "Vul uw naam in";
    }
    if (!values.emailAddress) {
      errors.emailAddress = "Vul uw e-mail adres in";
    } else if (!validateEmail(values.emailAddress)) {
      errors.emailAddress = "Geen valide e-mail adres";
    }

    return errors;
  },

  handleSubmit: (values, bag) => {
    bag.props.onSubmit(values);
  }
})(InnerForm);
