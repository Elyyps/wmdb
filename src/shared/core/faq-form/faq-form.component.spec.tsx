import React from "react";
import { shallow } from "enzyme";
import { FAQFormComponent } from "./faq-form.component";

describe("[FAQForm]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<FAQFormComponent onSubmit={() => ""} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
