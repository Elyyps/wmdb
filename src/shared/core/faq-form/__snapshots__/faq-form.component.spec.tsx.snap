// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[FAQForm] should render component without crashing 1`] = `
<Formik
  enableReinitialize={false}
  handleSubmit={[Function]}
  initialValues={
    Object {
      "emailAddress": "",
      "name": "",
    }
  }
  isInitialValid={false}
  onSubmit={[Function]}
  render={[Function]}
  validate={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
/>
`;
